prime-router/src/main/kotlin/credentials/AzureCredentialService.kt

```diff
@@ -0,0 +1,33 @@
+package gov.cdc.prime.router.credentials
+
+import com.azure.core.credential.TokenCredential
+import com.azure.identity.DefaultAzureCredentialBuilder
+import com.azure.security.keyvault.secrets.SecretClient
+import com.azure.security.keyvault.secrets.SecretClientBuilder
+
+object AzureCredentialService : CredentialService() {
+
```

Comment:
@RickHawesUSDS I agree on two points, but disagree on one of your points.

I agree that developers should not be using any of the service implementations directly, as dependency injections should be managing it for them. I also agree testability is a major concern with using singletons, and [while mockk does make testing singletons easier](https://notwoods.github.io/mockk-guidebook/docs/mocking/static/), it can still get messy.

What I disagree with is moving away from the singleton pattern. I think the issue here is visibility.

The credential service implementations should be marked as `internal` so they can not be instantiated outside of the package. If the implementations are marked as `internal`, the only way to use the credential service is through the [`CredentialManagement` interface](https://github.com/CDCgov/prime-data-hub/pull/392/files#diff-756129fa5f504f0c5fe47e9ccad012f0282f36c1e2aae91bf3910e691f96bf02) ([example of use](https://github.com/CDCgov/prime-data-hub/pull/392/files#diff-15b4e6a4398afcc6af7babd9fee78ef4cdf859897db923ceb34f465a5f844cecR20)).

The way `CredentialManagement` initializes the variable `credentialService` it's really easy to override in a test. Using the following syntax, you can quickly return a mock:

```
every { <your_class_here> getProperty "credentialService" } returns mockk<CredentialService>()
```

This was a great point to raise. I changed the visibility of the implementation classes in the following commit: https://github.com/CDCgov/prime-data-hub/pull/392/commits/78d3045e4ab16f2a7b484fd03b3a6672a00d2c98